### ISSUE TITLE
[Feature] Add the ability to mark commands and arguments as deprecated

### ISSUE THREAD

**Issue by skullydazed** [clueboard/milc#38]
Add a `deprecated=<reason>` argument to `@cli.entrypoint()`, `@cli.subcommand()`, and `@cli.argument()`. When present it will print `<reason>` when the command/argument is used, even when `--help` is passed.

**Comment by Fyrebright** [comment]
Hello, I was wondering if you intend this to be a keyword argument passed along with normal use of `@cli.argument()` so that it would look like:

```python
@cli.argument('-n', '--name', help='Name to greet', default='World', deprecated='entrypoint is deprecated')
@cli.entrypoint('Greet a user.')
def main(cli):
    ...
```
or are you expecting it to be put on its own as `@cli.argument(deprecated='entrypoint is deprecated')`?

Also, would it not make sense to put this in `@cli.entrypoint()`/`@cli.subcommand()` seeing as this is an attribute of the command itself?

**Comment by skullydazed** [comment]
Good questions! I jotted this down while I was working on something else, and didn't pay too much attention to the details.

I think you're right that adding it to `@cli.entrypoint()` and `@cli.subcommand()` make the most sense when deprecating an entire command. This brings up a related question- how does one deprecate an argument?

That would make the proper scope for this feature both commands and arguments. I'll update the issue match.

**Comment by Fyrebright** [comment]
I have created PR #39 with an example in the description. I wanted to do the helpstring by extending `HelpFormatter` but as far as I can tell, I would have to extend multiple classes in `argparse` which I thought would be more messy. I think adding the deprecation onto the help/description string achieves the same thing. Let me know if there are changes needed.

**Comment by Fyrebright** [comment]
I apologize for not following up on the review soon enough. I realized I had not done the CI tests properly the previous time and I was having trouble executing the properly on my Windows PC because WSL adds output to stdout when executed so the checks would fail. Do you use Cygwin for subshell? Or do you just run on a Linux OS?

**Comment by skullydazed** [comment]
No worries, I'm not working on any particular timeline here. :)

We use msys2, which is a cygwig derivative. What extra output is WSL adding?

**Comment by skullydazed** [comment]
To clarify, that's what the tests use for windows. We also test on ubuntu and macos. Maybe we need to add WSL to the matrix?

**Comment by skullydazed** [comment]
Implemented in #39. Thanks!

**Comment by Fyrebright** [comment]
@skullydazed It was adding the path header, but I realize I didn't have the Python env set up correctly, which fixed it. 
Glad I was able to contribute
